service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -877,127 +836,50 @@ private void setLocations(CreateAddressRequestDto addressRequestDto, Address add
-        GeocodingResult geocodingResult) {
-        initializedMap
-            .forEach((key, value) -> Arrays.stream(geocodingResult.addressComponents)
-                .forEach(addressComponent -> Arrays.stream(addressComponent.types)
-                    .filter(componentType -> componentType.equals(key))
-                    .forEach(componentType -> value.accept(addressComponent.longName))));
-    }
-
-    private OrderAddressDtoRequest getLocationDto(String placeId) {
-        GeocodingResult resultsUa = googleApiService.getResultFromGeoCode(placeId, 0);
-        GeocodingResult resultsEn = googleApiService.getResultFromGeoCode(placeId, 1);
-
-        OrderAddressDtoRequest orderAddressDtoRequest = new OrderAddressDtoRequest();
-        initializeGeoCodingResults(initializeUkrainianGeoCodingResult(orderAddressDtoRequest), resultsUa);
-        initializeGeoCodingResults(initializeEnglishGeoCodingResult(orderAddressDtoRequest), resultsEn);
-
-        double latitude = resultsEn.geometry.location.lat;
-        double longitude = resultsEn.geometry.location.lng;
-        orderAddressDtoRequest.setCoordinates(new Coordinates(latitude, longitude));
-
-        checkIfAddressBelongToKyiv(orderAddressDtoRequest);
-
-        return orderAddressDtoRequest;
+        return deletedAddress.orElse(null);
     }
 
     /**
-     * When Google API sets region name, there's a special case for Kyiv, the
-     * capital of Ukraine. Google API sets the 'Kyiv' as the name of region for
-     * addresses from Kyiv instead of Kyiv Region. Therefore, a separate logic is
-     * implemented to set 'Kyiv Region' name for such addresses instead of 'Kyiv'.
+     * Adjusts the region name for addresses in Kyiv to correctly reflect the 'Kyiv
+     * Region' instead of just 'Kyiv'. This is necessary because the Google API sets
+     * 'Kyiv' as the region name for addresses in Kyiv rather than 'Kyiv Region'.
+     * This method updates the region name for such addresses to 'Kyiv Region' to
+     * ensure consistency.
      *
-     * @param request OrderAddressDtoRequest.
+     * @param region  the {@link Region} object containing the name of the region
+     * @param address the {@link Address} object to be updated with the correct
+     *                region name
      */
-    private void checkIfAddressBelongToKyiv(OrderAddressDtoRequest request) {
-        if (request.getRegion().equalsIgnoreCase(KYIV_UA)) {
-            request.setRegion(KYIV_REGION_UA);
-        }
-
-        if (request.getRegionEn().equalsIgnoreCase(KYIV_EN)) {
-            request.setRegionEn(KYIV_REGION_EN);
-        }
-    }
-
     private void checkIfAddressBelongToKyiv(Region region, Address address) {
         if (region.getNameUk().equalsIgnoreCase(KYIV_UA)) {
             address.setRegion(KYIV_REGION_UA);
-            // todo: set Kyiv oblast region id
+            address.setRegionId(region);
         }
 
         if (region.getNameEn().equalsIgnoreCase(KYIV_EN)) {
```

Comment:
please use `KYIV_EN.equalsIgnoreCase(region.getNameEn())` to prevent NPE

Reply:
Very valid remark, I will take it into account.